### ISSUE TITLE
match arm formatting nits with multiple patterns

### ISSUE THREAD

**Issue by nrc** [rust-lang-nursery/rustfmt#1136]
From https://github.com/rust-lang/rust/pull/35850/files/ae74a1cb9e0430ade9045fa11ca223800c5a1560#r75770393

```
                 "packed" |
-                "simd" => {
-                    if target != Target::Struct {
-                        "attribute should be applied to struct"
-                    } else {
-                        continue
+                    "simd" => {
+                        if target != Target::Struct {
+                            "attribute should be applied to struct"
+                        } else {
+                            continue
+                        }
```

The pattern should not be line-broken (i.e., should be `"packed" | "simd" => {`)

The pattern is indented, and then the arm body is indented again. I think it would be better to keep the pattern indented to the same depth, and to only indent the arm body. However, if there is no arm body, but lots of patterns, then this is not great either, since there is no break between arms. So either we need a different heuristic there, or we need to insert a line break or something, or we eat this double indent.

**Comment by nikomatsakis** [comment]
> The pattern is indented, and then the arm body is indented again. I think it would be better to keep the pattern indented to the same depth, and to only indent the arm body. 

Yes

> However, if there is no arm body, but lots of patterns, then this is not great either, since there is no break between arms

What I do personally is either a line break or to put `=> { \n }`, to force some visual separation. A double indent seems pretty painful in comparison.

I do think it's important to have that visual separation!

Another option that I recall being popular from O'caml was to move the `|` at a roughly 2-space indent:

```rust
match foo {
    A
  | B
  | C => { }
    D
  | E
  | F => { }
}
```

but I'm inclined to think it's too clever by half. (Ocaml also allows you to put a leading `|` before the first pattern in each group, iirc, though, so I may be misremembering how people use it.)

**Comment by topecongiro** [comment]
Closed via #1775.
